hera_qm/tests/test_ant_metrics.py

```diff
@@ -0,0 +1,148 @@
+        metric3 = {(0,'x'): 1}
+        with self.assertRaises(KeyError):
+            ant_metrics.average_abs_metrics(metric1,metric3)
+
+class Test_Antenna_Metrics(unittest.TestCase):
+
+    def setUp(self):
+        dataFileList = [DATA_PATH + '/zen.2457698.40355.xx.HH.uvcA',
+                        DATA_PATH + '/zen.2457698.40355.xy.HH.uvcA',
+                        DATA_PATH + '/zen.2457698.40355.yx.HH.uvcA',
+                        DATA_PATH + '/zen.2457698.40355.yy.HH.uvcA']
+        reds = [[(9, 31), (20, 65), (22, 89), (53, 96), (64, 104), (72, 81), (112, 10), 
+                 (105, 20), (81, 43), (88, 53)], [(65, 72), (96, 105)], [(31, 105), 
+                 (43, 112), (96, 9), (65, 22), (89, 72), (104, 88)], [(20, 72), (43, 97), 
+                 (53, 105), (65, 81), (80, 88), (89, 112), (104, 9), (96, 20), (31, 22)], 
+                 [(22, 96), (72, 31), (112, 65), (105, 104)], [(9, 105), (10, 97), (20, 22), 
+                 (22, 72), (64, 88), (65, 89), (81, 112), (53, 9), (43, 10), (31, 20), (96, 31), 
+                 (104, 53), (80, 64), (89, 81)], [(96, 97), (104, 112), (80, 72)], [(10, 72), 
+                 (31, 88), (89, 105), (65, 9), (43, 22), (96, 64)], [(10, 105), (43, 9), (65, 64), 
+                 (89, 88)], [(9, 20), (20, 89), (22, 81), (31, 65), (72, 112), (80, 104), (88, 9), 
+                 (81, 10), (105, 22), (53, 31), (89, 43), (64, 53), (104, 96), (112, 97)], 
+                 [(31, 112), (53, 72), (65, 97), (80, 105), (104, 22), (96, 81)], [(72, 104), 
+                 (112, 96)], [(64, 97), (80, 10)], [(10, 64), (43, 80), (97, 88)], [(9, 80), 
+                 (10, 65), (20, 104), (22, 53), (89, 96), (72, 9), (112, 20), (81, 31), (105, 64), 
+                 (97, 89)], [(80, 112), (104, 97)], [(43, 105), (65, 88)], [(10, 22), (20, 88), 
+                 (31, 64), (81, 105), (89, 9), (43, 20), (65, 53), (97, 72), (96, 80)], [(43, 53), 
+                 (65, 80), (81, 88), (97, 105), (10, 9), (89, 64)], [(53, 97), (64, 112), (80, 81), 
+                 (104, 10)], [(9, 64), (10, 89), (20, 53), (31, 104), (43, 65), (53, 80), (65, 96), 
+                 (72, 105), (22, 9), (81, 20), (112, 22), (89, 31), (97, 81), (105, 88)], [(9, 112), 
+                 (20, 97), (53, 81), (31, 10), (80, 20), (64, 22), (96, 43), (88, 72), (104, 89)], 
+                 [(9, 81), (22, 97), (31, 43), (53, 89), (105, 112), (20, 10), (64, 20), (88, 22), 
+                 (80, 31), (104, 65)], [(43, 72), (65, 105), (96, 88)], [(31, 97), (53, 112), 
+                 (64, 72), (96, 10), (80, 22), (104, 81)], [(10, 88), (43, 64)], [(9, 97), (64, 81), 
+                 (80, 89), (88, 112), (53, 10), (104, 43)]]
+        self.am = ant_metrics.Antenna_Metrics(dataFileList, reds, fileformat='miriad')
+        with self.assertRaises(KeyError):
+            self.am.save_antenna_metrics(DATA_PATH + '/test_output/ant_metrics_output.json')
+        self.am.iterative_antenna_metrics_and_flagging()
```

Comment:
This setup is organized kind of funny. There shouldn't be an assert inside the setup, and I don't think the iterative function needs to be run (e.g. test_nPos doesn't need it.) I think it would make more sense to define `self.reds` and instantiate `self.am` in the tests when needed. Alternatively, is it possible to define an __init__ for this function that would run the tests just once? the setup gets run for each test.